Python/codecs.c

```diff
@@ -50,6 +50,27 @@ int PyCodec_Register(PyObject *search_function)
     return -1;
 }
 
+int
+PyCodec_Unregister(PyObject *search_function)
+{
+    PyInterpreterState *interp = PyInterpreterState_Get();
+    PyObject *codec_search_path = interp->codec_search_path;
+    /* Do nothing if codec_search_path is not created yet or was cleared. */
+    if (codec_search_path == NULL) {
+        return 0;
+    }
+
+    Py_ssize_t n = PyList_Size(codec_search_path);
```

Comment:
Please use the macro `PyList_GET_SIZE()` instead of `PyList_Size()`, as `codec_search_path` is guaranteed to be a list.

Reply:
While it's a list right now, its type can change tomorrow.

@shihai1991: If you modify the code to use fast macros, please add the following assertion *before* using it:

```
assert(PyList_CheckExact(interp->codec_search_path));
```

Reply:
@vstinner Copy that, I have already updated it.
But I have a question. `assert(PyList_Check(op)` in `_PyList_CAST()` is not good enough?